activation/wire/malfeasance_invalid_prev_atx.go

```diff
@@ -0,0 +1,231 @@
+package wire
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/spacemeshos/merkle-tree"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/signing"
+	"github.com/spacemeshos/go-spacemesh/sql"
+	"github.com/spacemeshos/go-spacemesh/sql/identities"
+)
+
+//go:generate scalegen
+
+// ProofInvalidPrevAtxV2 is a proof that two ATXs published by an identity reference the same previous ATX for an
+// identity.
+//
+// We are proving the following:
+// 1. The ATXs have different IDs.
+// 2. Both ATXs have a valid signature.
+// 3. Both ATXs reference the same previous ATX for the same identity.
+// 4. Both marriage certificates have valid signatures.
+//
+// HINT: this works if the identity that publishes the marriage ATX marries themselves.
```

Comment:
I think it's needed and enough to prove that the signers of the two ATXs are married.
```suggestion
// 4. Signers of bots ATXs are married
```

Reply:
I updated the description.